src/main/java/com/iota/iri/network/pipeline/PreProcessPayload.java

```diff
@@ -0,0 +1,52 @@
+package com.iota.iri.network.pipeline;
+
+import com.iota.iri.network.neighbor.Neighbor;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Defines the payload which gets submitted to the {@link PreProcessStage}.
+ */
+public class PreProcessPayload {
```

Comment:
1. See above, we can't express what we really want with generics. I also dropped `T` now from `ProcessingContext`, because it will just let warnings pop up with no real benefit, because we can't express the generality of the `process()` methods correctly. 
2. Payloads now extends the abstract class `Payload`